templates/hooks/archive/sdk_create_post_set_output.go.tpl

```diff
@@ -0,0 +1,5 @@
+if !archiveAvailable(&resource{ko}) {
+	ackcondition.SetSynced(&resource{ko}, corev1.ConditionFalse, nil, nil)
+} else {
+	ackcondition.SetSynced(&resource{ko}, corev1.ConditionTrue, nil, nil)
+}
```

Comment:
We already have this type of functionality built into the generator - to use statuses as the indicator for synced. Here is an example of how we do it for the `apigatewayv2-controller`: https://github.com/aws-controllers-k8s/apigatewayv2-controller/blob/main/generator.yaml#L98-L102

Reply:
Gotcha. So in all cases where I do not need to pass a message to `SetSynced`, e.g. `ackcondition.SetSynced(&resource{ko}, corev1.ConditionFalse, nil, nil)` I will remove this line and let code-gen figure out.

cc/ @A-Hilaly should also work in such cases then?

```go
if archiveInTerminalState(latest) {
	msg := "Archive is in '" + *latest.ko.Status.State + "' status"
	ackcondition.SetTerminal(desired, corev1.ConditionTrue, &msg, nil)
	ackcondition.SetSynced(desired, corev1.ConditionTrue, nil, nil) // TODO: remove
	return desired, nil
}
```

Using this `generator.yaml` syntax:

```yaml
    synced:
      when:
        - path: Status.State
          in:
            - ENABLED
            - DISABLED
            - CREATE_FAILED
            - UPDATE_FAILED
```

Reply:
Hm, not sure this one will work then?

```go
if archiveCreating(latest) {
	msg := "Archive is currently being created"
	ackcondition.SetSynced(desired, corev1.ConditionFalse, &msg, nil)
	return desired, requeueWaitUntilCanModify(latest)
}
```

Would code-gen `SYNCED` path respect my message here?

Reply:
I get the feeling to keep the logic as is for now as this change would require more testing (incl. E2Es) now to verify all the possible combinations again.

Reply:
I've noticed that when you use `is_synced` with other hooks, other components of the reconciliation logic can break. I've seen it happen before with lambda.
So, to keep things simple and speed up development for this controller, I'm cool with merging the current code as is and dealing with the issue in a separate PR for all the CRDs. That way, if the problem pops up with this controller, we'll know how to tackle it without getting bogged down in everything else.